keanu-python/keanu/optimization.py

```diff
@@ -0,0 +1,62 @@
+from py4j.java_gateway import java_import
+from keanu.base import JavaObjectWrapper
+from keanu.context import KeanuContext
+from keanu.inference import BayesNet
+from keanu.vertex import Vertex
+
+k = KeanuContext().jvm_view()
+
+java_import(k, "io.improbable.keanu.algorithms.variational.optimizer.gradient.GradientOptimizer")
+java_import(k, "io.improbable.keanu.algorithms.variational.optimizer.nongradient.NonGradientOptimizer")
+
+
+class Optimizer:
+    def __init__(self, optimizer, net):
+        self.optimizer = optimizer
+        self.net = net
+
+    def max_a_posteriori(self):
+        return self.optimizer.maxAPosteriori()
+
+    def max_likelihood(self):
+        return self.optimizer.maxLikelihood()
+
+    @staticmethod
+    def _build_bayes_net(builder, net):
```

Comment:
double underscore is preferred for private methods - it does name mangling.

https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-a-single-and-a-double-underscore-before-an-object-name

Reply:
As discussed offline - will leave this as a single underscore to avoid the nasty `_Optimizer__build_bayes_net`

Reply:
As discussed offline - will leave this as a single underscore to avoid the nasty `_Optimizer__build_bayes_net`